### PR TITLE
Set service.version on the service level

### PR DESCRIPTION
**Pull request by ghost**
### Which issue does this PR correspond to?
Fixes https://github.com/grapl-security/issue-tracker/issues/1051

### What changes does this PR make to Grapl? Why?
This sets the service.version tag at the service level instead of the job level. As Grapl only rebuilds and tags containers that have changes, the versions for each service within a job are not the same.

### How were these changes tested?
Tests

### CODE REVIEW

**Comment by twunderlich-grapl** [comment]
Right now I've chosen to create a second Mapping. An alternative approach (and more efficient) would be to create a single container_meta Mapping that looks something like
```
{
    container_name: {
        image: container_image_id
        version: container_version
    }
}
```

The tradeoff here is between verbosity (every affected line in Nomad would be longer) and efficiency

File: pulumi/infra/docker_images.py [hunk 1]
```diff
@@ -16,6 +16,18 @@
 - an image pulled from Cloudsmith
     "docker.cloudsmith.io/grapl/raw/graph-merger:20211105192234-a86a8ad2"
 """
+DockerImageVersion = NewType("DockerImageId", str)
```
Comment (christophermaier) [inline pulumi/infra/docker_images.py:19]: Should this be `NewType("DockerImageVersion", str)` instead?

File: nomad/grapl-core.nomad [hunk 2]
```diff
@@ -258,7 +263,7 @@ job "grapl-core" {
         RUST_BACKTRACE = local.rust_backtrace
         RUST_LOG       = var.rust_log
 
-        OTEL_RESOURCE_ATTRIBUTES = local.default_otel_resource_attributes
+        OTEL_RESOURCE_ATTRIBUTES = "${local.default_otel_resource_attributes},service.version=${var.container_versions["analyzer-dispatcher"]}"
```
Comment (twunderlich-grapl) [inline nomad/grapl-core.nomad:266]: Surprisingly the `"${var.foo["bar"}"` is the correct syntax. Single quotes are illegal in hcl2 and escaping quotes within interpolated values is not allowed

File: pulumi/grapl/__main__.py [hunk 3]
```diff
@@ -93,6 +93,15 @@ def _container_images(artifacts: ArtifactGetter) -> Mapping[str, DockerImageId]:
     }
 
 
+def _container_versions(
+    container_images: Mapping[str, DockerImageId]
+) -> Mapping[str, DockerImageVersion]:
+    container_versions = dict()
+    for key, value in container_images.items():
+        container_versions[key] = value.split(":")[1]
+    return container_versions
```
Comment (christophermaier) [inline pulumi/grapl/__main__.py:102]: It looks like this logic is the same in all the Pulumi projects; can we create a common implementation to cut down the repetition?

Comment (wimax-grapl) [inline pulumi/grapl/__main__.py:102]: slightly easier but not necessarily worth doing: a dict comprehension

```
return {
  key: value.split(":")[1]
  for key, value in container_images.items()
}
```

File: pulumi/infra/docker_images.py (hunk 1 above)
Comment (wimax-grapl) [inline pulumi/infra/docker_images.py:19]: change the string to match the type's name

File: pulumi/python_integration_tests/__main__.py [hunk 4]
```diff
@@ -70,10 +79,15 @@ def main() -> None:
         # on-disk Grapl repo.
         # FIXME: make python integration tests work in AWS.
 
+        container_images = (_python_integration_container_images(artifacts),)
```
Comment (wimax-grapl) [inline pulumi/python_integration_tests/__main__.py:82]: uhhhhh you sure about this? this makes container_images a tuple

File: pulumi/rust_integration_tests/__main__.py [hunk 5]
```diff
@@ -35,6 +35,15 @@ def _rust_integration_container_images(
...
 
+def _rust_integration_container_versions(
+    container_images: Mapping[str, DockerImageId]
+) -> Mapping[str, DockerImageVersion]:
+    container_versions = dict()
+    for key, value in container_images.items():
+        container_versions[key] = value.split(":")[1]
+    return container_versions
+
+
```
Comment (wimax-grapl) [inline pulumi/rust_integration_tests/__main__.py:46]: i feel like we could just centralize this on say docker_images.py

File: pulumi/infra/docker_images.py (hunk 1 above)
Comment (twunderlich-grapl) [reply pulumi/infra/docker_images.py]: Yes 🤦

File: pulumi/python_integration_tests/__main__.py (hunk 4 above)
Comment (twunderlich-grapl) [reply pulumi/python_integration_tests/__main__.py]: Ugh, no. I must have typed a paren by accident or soemthing here and my IDE autoexpanded. Thanks for catching this

File: pulumi/rust_integration_tests/__main__.py (hunk 5 above)
Comment (twunderlich-grapl) [reply pulumi/rust_integration_tests/__main__.py]: Done

**Review by christophermaier — APPROVED** [review]

**Review by wimax-grapl — APPROVED** [review]

**Pull request merged**